src/compiler/cpp_generator.cc

```diff
@@ -1272,7 +1274,6 @@ void PrintSourceService(Printer *printer, const Service *service,
 
   printer->Print(*vars, "$ns$$Service$::Service::Service() {\n");
   printer->Indent();
-  printer->Print(*vars, "(void)$prefix$$Service$_method_names;\n");
```

Comment:
The purpose of this line is to solve a similar problem. That is when there is no rpc methods defined in the service, some compilers will complain the global method_names array is not used. Since we do not generate them any more in that case, we do not need it. 

This is also the reason why I did not add NULL to the end of the array: so that we handle the case in one place rather than two. Not that really matters though.